packages/apple-pay-integration/src/apple-pay-payment-strategy.ts

```diff
@@ -267,29 +268,45 @@ export default class ApplePayPaymentStrategy implements PaymentStrategy {
     }
 
     private async _getBraintreeDeviceData() {
-        const data = await this._braintreeIntegrationService.getDataCollector();
+        let braintreePaymentMethod: PaymentMethod;
 
-        return data.deviceData;
+        try {
+            braintreePaymentMethod = this._paymentIntegrationService
+                .getState()
+                .getPaymentMethodOrThrow(ApplePayGatewayType.BRAINTREE);
+        } catch (_) {
+            return;
+        }
```

Comment:
It is not necessary to use try catch here. You can just remove orThrow.
```suggestion
            braintreePaymentMethod = this._paymentIntegrationService
                .getState()
                .getPaymentMethod(ApplePayGatewayType.BRAINTREE);
```